### PR TITLE
ci: Add Arm skip option to tests

### LINKED ISSUE #2034: Tests with autotune.py fail on m1
A selection of tests are failing when ran on a parallel ubuntu desktop running on mac m1:

tests/test_autotuner.py .................F.                                                                                                            [  1%]
...
tests/test_buffering.py ..............x.........                                                                                                       [  3%]
...
tests/test_data.py .............................ss...                                                                                                  [ 11%]
...
tests/test_dse.py ......x............................................................................................................................. [ 47%]
.........FF............F............................................                                                                                   [ 51%]

Steps to reproduce problem:
1. run Linux ubuntu-linux-22-04-desktop 5.15.0-41-generic on parallel desktop
2. Install gcc python according to the version outlined below
3. python3 -m -venv  ~/env/devito_env
4. cd work_directory
5. source ~/env/devito_env/bin/activate
6. git clone https://github.com/devitocodes/devito.git
7. cd devito
8. pip install -e .
9. pytest -m "not parallel" -k "not adjoint" tests/

Host Hardware Description:
2021 14 inch Mac book Pro Apple M1 Max
Host OS Description:
Ventura 13.0

Virtual Environment Set-up:
Linux ubuntu-linux-22-04-desktop 5.15.0-41-generic #44-Ubuntu SMP Thu Jun 23 11:20:13 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Python 3.10.6

Application used for virtualisation:
Parallels Desktop

### PR DESCRIPTION
**Pull request by mloubout**
Fixes #2034

### CODE REVIEW

**Review by georgebisbas — COMMENTED** [review]
In general, these tests do not fail on all other arm platforms (e.g. a64fx).
Though I admit, they are not on CI.

**Comment by mloubout** [comment]
>o not fail on all other arm platforms 

Are we sure about this? I would assume depends on the compiler

File: tests/test_dse.py [hunk 1]
```diff
@@ -1626,7 +1627,7 @@ def test_hoisting_symbolic_divs(self):
         op = Operator(eq)
 
         assert op._profiler._sections['section0'].sops == 1
-        assert op.body.body[1].body[0].body[0].expr.rhs == s0**-s1
```
Comment (FabioLuporini) [inline tests/test_dse.py:1629]: why this? I suppose with Arm a slightly different code gets generated... is it the lack of the denormals flag?

**Comment by FabioLuporini** [comment]
Uncontroversial

File: tests/test_dse.py (hunk 1 above)
Comment (mloubout) [reply tests/test_dse.py:1629]: Yes the [1] assumes there is the denormal body first so breaks on non-sse

**Pull request merged**